00.prerequisites/00.ad.documentation.md

```diff
@@ -0,0 +1,105 @@
+Collecting click>=8.1.3
+  Downloading click-8.1.6-py3-none-any.whl (97 kB)
+     |████████████████████████████████| 97 kB 2.2 MB/s 
+Collecting blinker>=1.6.2
+  Downloading blinker-1.6.2-py3-none-any.whl (13 kB)
+Collecting Werkzeug>=2.3.3
+  Downloading Werkzeug-2.3.6-py3-none-any.whl (242 kB)
+     |████████████████████████████████| 242 kB 2.1 MB/s 
+Collecting MarkupSafe>=2.0
+  Downloading MarkupSafe-2.1.3-cp310-cp310-macosx_10_9_x86_64.whl (13 kB)
+Installing collected packages: MarkupSafe, Werkzeug, Jinja2, itsdangerous, click, blinker, flask
+Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.3 Werkzeug-2.3.6 blinker-1.6.2 click-8.1.6 flask-2.3.2 itsdangerous-2.1.2`
+
+Installed Python packages:
+
+`Andreas-MacBook-Pro:RESTing-with-Flask andreadashe$ pip3 install argparse
+Collecting argparse
+  Downloading argparse-1.4.0-py2.py3-none-any.whl (23 kB)
+Installing collected packages: argparse
+Successfully installed argparse-1.4.0`
+
+Created a virtual environment:
+
+`Andreas-MacBook-Pro:RESTing-with-Flask andreadashe$ python3 -m venv venv`
+
+`Andreas-MacBook-Pro:RESTing-with-Flask andreadashe$ source ./venv/bin/activate
+(venv) `
+
+Installed Flask requests:
+
+`Andreas-MacBook-Pro:RESTing-with-Flask andreadashe$ pip3 install flask requests
+Collecting flask
+  Using cached Flask-2.3.2-py3-none-any.whl (96 kB)
+Collecting requests
+  Downloading requests-2.31.0-py3-none-any.whl (62 kB)
+     |████████████████████████████████| 62 kB 1.2 MB/s 
+Collecting Werkzeug>=2.3.3
+  Using cached Werkzeug-2.3.6-py3-none-any.whl (242 kB)
+Collecting blinker>=1.6.2
+  Using cached blinker-1.6.2-py3-none-any.whl (13 kB)
+Collecting Jinja2>=3.1.2
+  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
+Collecting itsdangerous>=2.1.2
+  Using cached itsdangerous-2.1.2-py3-none-any.whl (15 kB)
+Collecting click>=8.1.3
+  Using cached click-8.1.6-py3-none-any.whl (97 kB)
+Collecting charset-normalizer<4,>=2
+  Downloading charset_normalizer-3.2.0-cp310-cp310-macosx_10_9_x86_64.whl (126 kB)
+     |████████████████████████████████| 126 kB 1.9 MB/s 
+Collecting urllib3<3,>=1.21.1
+  Downloading urllib3-2.0.4-py3-none-any.whl (123 kB)
+     |████████████████████████████████| 123 kB 2.3 MB/s 
+Collecting idna<4,>=2.5
+  Downloading idna-3.4-py3-none-any.whl (61 kB)
+     |████████████████████████████████| 61 kB 143 kB/s 
+Collecting certifi>=2017.4.17
+  Downloading certifi-2023.7.22-py3-none-any.whl (158 kB)
+     |████████████████████████████████| 158 kB 2.0 MB/s 
+Collecting MarkupSafe>=2.0
+  Using cached MarkupSafe-2.1.3-cp310-cp310-macosx_10_9_x86_64.whl (13 kB)
+Installing collected packages: MarkupSafe, Werkzeug, urllib3, Jinja2, itsdangerous, idna, click, charset-normalizer, certifi, blinker, requests, flask
+Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.3 Werkzeug-2.3.6 blinker-1.6.2 certifi-2023.7.22 charset-normalizer-3.2.0 click-8.1.6 flask-2.3.2 idna-3.4 itsdangerous-2.1.2 requests-2.31.0 urllib3-2.0.4`
+
+Saved requirements.txt:
```

Comment:
Note that your `requirements.txt` now has all the component versions fixed. 
You may want tot hink about benefits and problems this may cause....